explainer.md

```diff
@@ -182,6 +182,50 @@ The main benefit to this approach is that it would ensure that developers don't
 
 As an aside, the `notificationclick` event has similar challenges to the `launch` event in that handlers can be written such that nothing happens when a notification is clicked. Whatever solution is decided for `launch` event should also apply to `notificationclick` for consistency.
 
+### Should apps that aren't installed have launch events fired?
+
+It is unclear whether launch events should be gated on installation. Ultimately this should be up to the user agent, but it would be nice to be able to provide some kind of recommendation. 
+
+Our preference here is that apps which are not installed don't receive launch events, so as to break user expectations of the web as little as possible (installed apps are a new thing). In general, we prefer not to gate APIs on installation, however, there is some precedence for granting more permissions upon installation; PiP and WebShareTarget, for example.
+
+### Potentially Breaking `window.open`
+`window.open` is a synchronous  API, while the nature of launch events requires them to be asynchronous. It is unclear how we will be able to reconcile these two APIs (blocking until the launch event completes doesn't seem super acceptable). This could break popup based authentication flows (and potentially more).
+
+Due to this concern, the initial implementation of launch events will likely ship without `window.open` support, with the intent to revisit it later.
```

Comment:
It could be good to have an idea about how it would work before launch just to make sure it is possible in the current design :)

Reply:
I'm actually thinking maybe blocking is fine (as we intend to have a pretty agressive timeout). We could begin warming (loading?) a new window in the background and show it if it turns out that's what we should've been doing, and destroy it if the launch event wants us to do something else. What do you think? Is this too scary?

Reply:
See the note above - we can chat more in the office about it though :)